pkg/meta/sql.go

```diff
@@ -800,6 +800,9 @@ func (m *dbMeta) shouldRetry(err error) bool {
 		logger.Warnf("transaction failed: %s, will retry it. please increase the max number of connections in your database, or use a connection pool.", msg)
 		return true
 	}
+	if strings.Contains(msg, "error 1020 (hy000)") {
```

Comment:
Better add explanation about this error. Is it a common error or mariadb specific?